.travis.yml

```diff
@@ -21,20 +21,37 @@ cache:
 before_install:
   - npm install -g npm@6.14.7
   - sudo apt-get install libgconf2-4
+  - export SOURCE_BRANCH=${TRAVIS_PULL_REQUEST_BRANCH:-$TRAVIS_BRANCH}
+  - export IS_FORKED_PULL_REQUEST=$([[ $TRAVIS_PULL_REQUEST != false && $TRAVIS_PULL_REQUEST_SLUG != $TRAVIS_REPO_SLUG ]] && echo true || echo false)
+  - export HAS_BETA_TAG=$([[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+-beta$ && $IS_FORKED_PULL_REQUEST = false ]] && echo true || echo false)
+  - export HAS_RELEASE_TAG=$([[ $TRAVIS_TAG =~ ^[0-9]+\.[0-9]+\.[0-9]+$ && $IS_FORKED_PULL_REQUEST = false ]] && echo true || echo false)
+  - export IS_ON_FEATURE_BRANCH=$([[ $SOURCE_BRANCH =~ ^JSUI-[0-9]+ && $IS_FORKED_PULL_REQUEST = false ]] && echo true || echo false)
+  - export IS_NIGHTLY=$([[ $HAS_BETA_TAG = true && $TRAVIS_EVENT_TYPE = cron ]] && echo true || echo false)
+  - export IS_PULL_REQUEST_PUSH_BUILD=$([[ $TRAVIS_PULL_REQUEST = false && $IS_ON_FEATURE_BRANCH = true ]] && echo true || echo false)
```

Comment:
This is only `true` on `continuous-integration/travis-ci/push` builds.

Reply:
Looking back at this, I'm realizing that this would likely cause push builds to be triggered on `JSUI-*` branches, whether they have a PR or not. However, for certain features like QuestionAnswering (JSUI-2049) that don't necessarily have an open PR, this could be beneficial.

@samisayegh @ThibodeauJF any opinion?

Reply:
I think pushing all `JSUI-*` branches is fine. Let's start with this. If we see it is too for some reason, we can adjust.